src/pages/workspaces/workspace/Workflows.js

```diff
@@ -189,20 +201,60 @@ const FindWorkflowModal = ajaxCaller(class FindWorkflowModal extends Component {
 
     const { synopsis, managers, documentation } = selectedWorkflowDetails || {}
 
+    const errors = validateWorkflowName(editedWorkflowName, this.props.name)
+
     const renderDetails = () => [
       div({ style: { display: 'flex' } }, [
-        div({ style: { flexGrow: 1 } }, [
-          div({ style: { fontSize: 18, fontWeight: 600, margin: '1rem 0 0.5rem' } }, ['Synopsis']),
+        div({ style: { flexGrow: 1, marginTop: '1rem' } }, [
+          !isEditingName ?
+            div([span({ style: styles.sectionTitle }, ['Workflow Name: ']), workflowRename ? `${workflowRename}` : '',
+              h(Link, {
+                style: { marginRight: '0.5rem' },
+                onClick: () => this.setState({ isEditingName: !isEditingName }),
+                'aria-label': 'Edit workflow name'
+              }, [workflowRename ? icon('edit', { style: { marginLeft: '0.5rem' }, size: 13 }) : undefined])])
+            :
+            h(Fragment, [
+              h(IdContainer, [id => h(Fragment, [
+                h(FormLabel, { htmlFor: id, style: { margin: '0 0 0.25rem' } }, ['Workflow Name:']),
+                h(ValidatedInput, {
+                  error: Utils.summarizeErrors(errors),
+                  inputProps: {
+                    id,
+                    value: editedWorkflowName,
+                    onChange: editedWorkflowName => { this.setState({ editedWorkflowName }) }
+                  }
+                })
+              ])]),
+              div({ style: { marginTop: '0.5rem' } }, [
+                h(ButtonPrimary, {
+                  style: { marginRight: '0.5rem' },
+                  disabled: !!errors,
+                  onClick: () => this.setState({ isEditingName: !isEditingName, workflowRename: editedWorkflowName })
+                }, ['save']),
+                h(ButtonSecondary, {
+                  onClick: () => this.setState({ isEditingName: !isEditingName, editedWorkflowName: workflowRename })
+                }, ['cancel'])
+              ])
+            ]),
+          div({ style: styles.sectionTitle }, ['Synopsis']),
           div([synopsis || (selectedWorkflowDetails && 'None')]),
-          div({ style: { fontSize: 18, fontWeight: 600, margin: '1rem 0 0.5rem' } }, ['Method Owner']),
+          div({ style: styles.sectionTitle }, ['Method Owner']),
           div([_.join(',', managers)])
         ]),
         div({ style: { margin: '0 1rem', display: 'flex', flexDirection: 'column' } }, [
-          h(ButtonPrimary, { style: { marginBottom: '0.5rem' }, onClick: () => this.exportMethod() }, ['Add to Workspace']),
-          h(ButtonOutline, { onClick: () => this.setState({ selectedWorkflow: undefined, selectedWorkflowDetails: undefined }) }, ['Return to List'])
+          h(ButtonPrimary, {
+            disabled: !!errors || !!isEditingName,
```

Comment:
nit: Since `isEditingName` is already a boolean, there's no need to use `!!` on it.